### PR TITLE
allow paths with periods in the middle, but not with leading slash

### PR DESCRIPTION
**Pull request by ssemakov**
Hey,

this is about multiple dots in URL problem. 
I know you've been working on this before (https://github.com/eliotsykes/rack-zippy/issues/17), but  I want to propose a bit modified solution that I've created for myself. It allows paths like /something...with..dots/ but still respond 404 on paths like /.. or /..something

Thanks,
-Simon

### CODE REVIEW

**Comment by eliotsykes** [comment]
Thanks for the contribution @ssemakov. I'd like to merge this improvement in. I'm shortly going to add some minor feedback to the PR that'd be great if you could address.

File: lib/rack-zippy.rb [hunk 1]
```diff
@@ -63,7 +63,7 @@ def asset_root
 
       ACCEPTS_GZIP_REGEX = /\bgzip\b/
 
-      ILLEGAL_PATH_REGEX = /(\.\.|\/\.)/
+      ILLEGAL_PATH_REGEX = /(\/\.\.|\/\.)/
```
Comment (eliotsykes) [inline lib/rack-zippy.rb]: Can this be simplified to the following or similar?
```
ILLEGAL_PATH_REGEX = /(\/\.\.?)/
```

File: test/asset_server_test.rb [hunk 2]
```diff
@@ -198,6 +198,13 @@ def test_responds_not_found_if_path_contains_consecutive_periods
         end
       end
 
+      def test_responds_ok_if_path_contains_periods_that_not_follow_slash
+         ["/hello/path..with....periouds/file", "/hello/path/with..periods"].each do |dotty_path|
```
Comment (eliotsykes) [inline test/asset_server_test.rb]: Minor typo: periouds -> periods

File: test/asset_server_test.rb [hunk 3]
```diff
@@ -198,6 +198,13 @@ def test_responds_not_found_if_path_contains_consecutive_periods
...
+          get dotty_path
+          assert_response_ok
```
Comment (eliotsykes) [inline test/asset_server_test.rb]: Consider using `assert_underlying_app_responded` instead of `assert_response_ok`

File: lib/rack-zippy.rb (hunk 1 above)
Comment (ssemakov) [inline lib/rack-zippy.rb]: Yeah, this looks better.

**Comment by eliotsykes** [comment]
Thanks @ssemakov - released in 3.0.1

**Pull request merged**